Assets/Managers/pipeManager.cs

```diff
@@ -70,27 +72,33 @@ void Update()
 
             //This section deals with transerving out all possible ends of the pipe that is not the entry side
             foreach(Vector2 element in connectingPoints){
-                //Debug.Log("STAGE 1");
-                //Debug.Log($"checking point {element}");
                 if(element != connectingPostion && connectingPostion != Vector2.zero){
-                    Debug.Log("STAGE 2");
                     Debug.Log($"chosen point {element}");
                     if(waterControl.canMoveDirection(element)){
-                        currentPiece.GetComponent<SpriteRenderer>().color = Color.blue;
-                        tranverse(element,currentPos);
+                        if(connectingPoints.Length > 2 && currentPiece.GetComponent<pipe>().returnIsBalanceSplitter()){
```

Comment:
I support this logic. A reminder that animations and delays will replace the Color.blue condition change. Perfect placeholder for now.